### PR TITLE
Improve syncing docs

### PR DESCRIPTION
**Pull request by nwalters512**
This PR makes a number of improvements to the syncing docs:

- Documents the fact that in-browser editing can be used
- Documents a preferred GitHub workflow
- Discusses the tradeoffs and benefits of different methods of editing course content
- Removes redundant information about permissions, which can be found on the "Course configuration" page instead.

### CODE REVIEW

File: docs/sync.md [hunk 1]
```diff
@@ -1,50 +1,35 @@
-# Syncing course content to the live website
+# Syncing course content
```
Comment (nwalters512) [inline docs/sync.md:1]: I'm inclined to rename this page to "Syncing and editing course content"?

Comment (mwest1066) [reply docs/sync.md:1]: Sounds good to me!

**Review by mwest1066 — APPROVED** [review]

File: docs/sync.md (hunk 1 above)
Comment (nwalters512) [reply docs/sync.md]: Done in 024e52f!

File: docs/sync.md [hunk 2]
```diff
@@ -1,50 +1,33 @@
...
 
-| Level  | Permissions                                                                                |
-| ------ | ------------------------------------------------------------------------------------------ |
-| Owner  | Can sync content, edit in the browser, and add/remove course permissions for other people. |
-| Editor | Can sync content, edit in the browser, but cannot change other people's permissions.       |
-| Viewer | Can view course level content (questions and issues), but can't edit or sync content.      |
+We recommend the [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow): users should clone the course repository, check out a new branch, push their changes to the branch, and open a pull request. Once their changes have been reviewed, approved, and merged, an instructor with "Editor" permissions in the course can navigate to the "Sync" page of the course and sync the changes into PrairieLearn with the "Pull from remote git repository" button. Note that your must have "Editor" or "Owner" permissions in the course to edit files.
```
Comment (nwalters512) [inline docs/sync.md:25]:
```suggestion
We recommend the [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow): users should clone the course repository, check out a new branch, push their changes to the branch, and open a pull request. Once their changes have been reviewed, approved, and merged, an instructor with "Editor" permissions in the course can navigate to the "Sync" page of the course and sync the changes into PrairieLearn with the "Pull from remote git repository" button. Note that your must have "Editor" or "Owner" permissions in the course to sync the repository.
```

**Pull request merged**